File: src/main/java/PalindromeNumber.java
public class PalindromeNumber {
    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }
        int reversed = reverse(x);
        return x == reversed;
    }

    private int reverse(int x) {
        long output = 0;
        int currentDigit;
        while (x != 0) {
            currentDigit = x % 10;
            x = x / 10;
            output = output * 10 + currentDigit;
            if (output > Integer.MAX_VALUE || output < Integer.MIN_VALUE) {
                return 0;
            }
        }
        return (int) output;
    }
}

File: src/main/kotlin/com/riteshpuj/primitives/PalindromeNumber.kt
package com.riteshpuj.primitives

class PalindromeNumber {
    fun isPalindrome(x: Int): Boolean {
        if(x < 0) return false
        var reverse = 0
        var current = x
        while(current != 0){
            reverse = current % 10 + reverse * 10
            current /= 10
        }
        return reverse == x
    }
}